packages/fluentui/react-northstar/src/components/Datepicker/DatepickerCalendar.tsx

```diff
@@ -210,19 +250,13 @@ export const DatepickerCalendar: ComponentWithAs<'div', DatepickerCalendarProps>
         break;
     }
 
-    if (!targetDate) {
-      // if we couldn't find a target date at all, do nothing
-      return;
-    }
-    const newNavigateDate = findAvailableDate({
-      initialDate,
-      targetDate,
-      direction,
-    });
-    if (!dayInGrid(visibledGrid, newNavigateDate)) {
-      setGridNavigatedDate(newNavigateDate);
-      e.preventDefault();
+    const newNavigatedDate = contstraintNavigatedDate(initialDate, targetDate, direction);
+
+    if (newNavigatedDate) {
+      setGridNavigatedDate(newNavigatedDate);
     }
+
+    e.preventDefault();
```

Comment:
Should we prevent always, or only when we are actively change focus (i.e. inside the last `if` clause)?

Reply:
The other case is that we don't have anywhere in the cell grid to move. Which I believe for that case, we would not also want some default behavior to overtake, or?